openedx_tagging/core/tagging/api.py

```diff
@@ -0,0 +1,136 @@
+        if changed:
+            object_tag.save()
+            num_changed += 1
+    return num_changed
+
+
+def get_object_tags(
+    taxonomy: Taxonomy, object_id: str, object_type: str, valid_only=True
+) -> List[ObjectTag]:
+    """
+    Returns a list of tags for a given taxonomy + content.
+
+    Pass valid_only=False when displaying tags to content authors, so they can see invalid tags too.
+    Invalid tags will likely be hidden from learners.
+    """
+    tags = ObjectTag.objects.filter(
+        taxonomy=taxonomy, object_id=object_id, object_type=object_type
+    )
+    return [tag for tag in tags if not valid_only or taxonomy.validate_object_tag(tag)]
+
+
+def tag_object(
+    taxonomy: Taxonomy, tags: List, object_id: str, object_type: str
+) -> List[ObjectTag]:
+    """
+    Replaces the existing ObjectTag entries for the given taxonomy + object_id with the given list of tags.
+
+    If taxonomy.allows_free_text, then the list should be a list of tag values.
+    Otherwise, it should be a list of existing Tag IDs.
+
+    Raised ValueError if the proposed tags are invalid for this taxonomy.
+    Preserves existing (valid) tags, adds new (valid) tags, and removes omitted (or invalid) tags.
+    """
+
+    return taxonomy.tag_object(tags, object_id, object_type)
+
+
+def autocomplete_tags(taxonomy: Taxonomy, prefix: str, object_id: str= None, count=10) -> List[str]:
+    """
+    Returns the first `count` tag values in the given Taxonomy with names 
+    that begin with the given prefix string. The result is returned in alphabetical
+
+    Closed taxonomies return tag values that exist in the taxonomy. Also excludes all Tags that the
+    `object_id` already has. 
+
+    Free-text taxonomies return only tag values that are currently in use on (another) object. Also excludes
+    the tags used by `object_id`.
+    """
+    result = []
+    excluded_tags = ObjectTag.objects.filter(object_id=object_id, taxonomy=taxonomy)
+    if taxonomy.allow_free_text:
+        # Free-text taxonomy
+        excluded_tags = excluded_tags.values_list('_value', flat=True)
+        result = ObjectTag.objects.filter(taxonomy=taxonomy, _value__startswith=prefix) \
+                                  .exclude(object_id=object_id) \
+                                  .exclude(_value__in=excluded_tags) \
+                                  .order_by('_value') \
+                                  .values_list('_value', flat=True)
+        # Remove repeat tags
+        result = list(set(result))[:count]
+    else:
+        # Closed taxonomy
+        excluded_tags = excluded_tags.filter(tag__isnull=False).values_list('tag__id', flat=True)
+        result = Tag.objects.filter(taxonomy=taxonomy, value__startswith=prefix) \
+                            .exclude(id__in=excluded_tags) \
+                            .order_by('value') \
+                            .values_list('value', flat=True) \
+                            [:count]
```

Comment:
Should do the same here just in case there's repeated tag values in the taxonomy:

```suggestion
                            .values_list('value', flat=True) \
                            .distinct()[:count]
```